--- lynx2-8-8-orig/src/LYMain.c
+++ lynx2-8-8/src/LYMain.c
@@ -1486,6 +1486,13 @@
 	    lynx_cfg_file = NULL;
 	}
     }
+    if (!lynx_cfg_file) {
+	HTSprintf0(&lynx_cfg_file, "%s\\..\\etc\\lynx.cfg", program_dir);
+	if (!LYCanReadFile(lynx_cfg_file)) {
+	    FREE(lynx_cfg_file);
+	    lynx_cfg_file = NULL;
+	}
+    }
 #endif
 
     /*
--- lynx-orig/src/LYStyle.c
+++ lynx/src/LYStyle.c
@@ -922,7 +922,7 @@
 	       (cp = LYGetEnv("lynx_lss")) != NULL) {
 	lynx_lss_file = find_lss_file(cp);
     } else {
-	lynx_lss_file = find_lss_file(cp = DeConst(LYNX_LSS_FILE));
+	HTSprintf0(&lynx_lss_file, "%s\\..\\etc\\lynx.lss", program_dir);
     }
     CTRACE1((tfp, "init_color_styles(%s)\n", NonNull(lynx_lss_file)));
 
