--- lynx2-8-8-orig/src/LYMain.c
+++ lynx2-8-8/src/LYMain.c
@@ -1486,6 +1486,13 @@
 	    lynx_cfg_file = NULL;
 	}
     }
+    if (!lynx_cfg_file) {
+	HTSprintf0(&lynx_cfg_file, "%s\\..\\etc\\lynx.cfg", program_dir);
+	if (!LYCanReadFile(lynx_cfg_file)) {
+	    FREE(lynx_cfg_file);
+	    lynx_cfg_file = NULL;
+	}
+    }
 #endif
 
     /*
--- lynx-orig/src/LYUtils.c
+++ lynx/src/LYUtils.c
@@ -2895,6 +2895,13 @@
 		    LYTildeExpand(&result, TRUE);
 		    if (!LYCanReadFile(result)) {
 			FREE(result);
+                        StrAllocCopy(result, program_dir);
+                        StrAllocCat(result, "\\..\\etc\\");
+                        StrAllocCat(result, nominal);  printf("trying -%s-\n", result);
+                        if (!LYCanReadFile(result))
+                        {
+                            FREE(result);
+                        }
 		    }
 		}
 #endif
