Use the right Python headers.
--- libieee1284-0.2.11-orig/Makefile.am
+++ libieee1284-0.2.11/Makefile.am
@@ -36,7 +37,7 @@
 pyexec_LTLIBRARIES = src/ieee1284module.la
 src_ieee1284module_la_LDFLAGS = -module -avoid-version
 src_ieee1284module_la_LIBADD = libieee1284.la
-src_ieee1284module_la_CFLAGS = -I/usr/include/python$(PYTHON_VERSION) -fpic
+src_ieee1284module_la_CFLAGS = -I$(PYTHON_PREFIX)/include/python$(PYTHON_VERSION) -fpic
 endif

 man3_MANS = doc/libieee1284.3 \
