diff -u -r --new-file C++/my_sal.h C++fix/my_sal.h
--- C++/my_sal.h	1969-12-31 16:00:00.000000000 -0800
+++ C++fix/my_sal.h	2015-04-22 21:36:46.930542600 -0700
@@ -0,0 +1,26 @@
+#pragma once
+#define _In_z_
+#define _Inout_z_
+#define _In_reads_(s)
+#define _In_reads_bytes(s)
+#define _In_reads_z_(s)
+#define _In_reads_or_z_(s)
+#define _Out_writes_(s)
+#define _Out_writes_bytes_(s)
+#define _Out_writes_z_(s)
+#define _Inout_updates_(s)
+#define _Inout_updates_bytes_(s)
+#define _Inout_updates_bytes_(s)
+#define _Inout_updates_z_(s)
+#define _Out_writes_to_(s,c)
+#define _Out_writes_bytes_to_(s,c)
+#define _Out_writes_bytes_to_(s,c)
+#define _Out_writes_all_(s)
+#define _Out_writes_bytes_all_(s)
+#define _Inout_updates_to_(s,c)
+#define _Inout_updates_bytes_to_(s,c)
+#define _Inout_reads_to_ptr_(p)
+#define _In_reads_to_ptr_(p)
+#define _In_reads_to_ptr_z_(p)
+#define _Out_writes_to_ptr_(p)
+#define _Out_writes_to_ptr_z(p)
diff -u -r --new-file C++/uvcview.h C++fix/uvcview.h
--- C++/uvcview.h	2015-04-22 21:35:58.140666500 -0700
+++ C++fix/uvcview.h	2015-04-22 21:36:30.876008500 -0700
@@ -24,6 +24,7 @@
 /*****************************************************************************
  I N C L U D E S
 *****************************************************************************/
+#include "my_sal.h"
 #include <windows.h>
 #include <windowsx.h>
 #include <initguid.h>
