--- fceux-2.2.2-orig/SConstruct 2015-07-19 22:22:25.190873400 -0700
+++ fceux-2.2.2/SConstruct      2015-07-19 22:22:36.659528800 -0700
@@ -107,8 +107,8 @@
       Exit(1)
     env.ParseConfig('sdl-config --cflags --libs')
   if env['GTK']:
-    if not conf.CheckLib('gtk-x11-2.0'):
-      print 'Could not find libgtk-2.0, exiting!'
+    if not (conf.CheckLib('gtk-x11-2.0') or conf.CheckLib('gtk-win32-2.0')):
+      print 'Could not find libgtk-2.0, exiting!!'
       Exit(1)
     # Add compiler and linker flags from pkg-config
     config_string = 'pkg-config --cflags --libs gtk+-2.0'
