diff -u -r fceux-2.2.2-orig/src/version.h fceux-2.2.2/src/version.h
--- fceux-2.2.2-orig/src/version.h	2015-09-15 08:31:32.163463400 -0700
+++ fceux-2.2.2/src/version.h	2015-09-15 08:31:39.365742500 -0700
@@ -19,8 +19,6 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
  */
 
-#define PUBLIC_RELEASE		// uncomment this when making a public release, but comment back before committing
-
 #ifndef __FCEU_VERSION
 #define __FCEU_VERSION
 
