--- fceux-2.2.2-orig/src/SConscript     2015-07-19 23:16:12.859844200 -0700
+++ fceux-2.2.2/src/SConscript  2015-07-19 23:16:23.463055200 -0700
@@ -27,7 +27,7 @@
 for dir in subdirs:
   subdir_files = SConscript('%s/SConscript' % dir)
   file_list.append(subdir_files)
-if env['PLATFORM'] == 'win32':
+if env['PLATFORM'] == 'win32' or env['PLATFORM'] == 'msys':
   platform_files = SConscript('drivers/win/SConscript')
 else:
   platform_files = SConscript('drivers/sdl/SConscript')
