--- fceux-2.2.2-orig/src/drivers/win/SConscript 2015-07-21 09:07:29.640918200 -0700
+++ fceux-2.2.2/src/drivers/win/SConscript      2015-07-21 09:08:08.670295400 -0700
@@ -48,9 +48,7 @@
 my_list.append('res.o')
 env.Command('res.o', 'res.rc', env['WINDRES'] + ' -Isrc/drivers/win -DLVS_OWNERDATA=0x1000 -o $TARGET $SOURCE')

-subdirs = Split("""
-directx
-zlib""")
+subdirs = []

 for x in range(len(my_list)):
   my_list[x] = 'drivers/win/' + my_list[x]
