--- fceux-2.2.2-orig/SConstruct 2015-07-21 09:20:32.134600000 -0700
+++ fceux-2.2.2/SConstruct      2015-07-21 09:32:26.078646200 -0700
@@ -187,7 +187,7 @@

 Export('env')
 fceux = SConscript('src/SConscript')
-env.Program(target="fceux-net-server", source=["fceux-server/server.cpp", "fceux-server/md5.cpp", "fceux-server/throttle.cpp"])
+

 # Installation rules
 if prefix == None:
@@ -200,8 +200,8 @@
 fceux_src = 'src/fceux' + exe_suffix
 fceux_dst = 'bin/fceux' + exe_suffix

-fceux_net_server_src = 'fceux-net-server' + exe_suffix
-fceux_net_server_dst = 'bin/fceux-net-server' + exe_suffix
+
+

 auxlib_src = 'src/auxlib.lua'
 auxlib_dst = 'bin/auxlib.lua'
@@ -212,13 +212,13 @@

 env.Command(fceux_h_dst, fceux_h_src, [Copy(fceux_h_dst, fceux_h_src)])
 env.Command(fceux_dst, fceux_src, [Copy(fceux_dst, fceux_src)])
-env.Command(fceux_net_server_dst, fceux_net_server_src, [Copy(fceux_net_server_dst, fceux_net_server_src)])
+
 env.Command(auxlib_dst, auxlib_src, [Copy(auxlib_dst, auxlib_src)])

 man_src = 'documentation/fceux.6'
-man_net_src = 'documentation/fceux-net-server.6'
+
 man_dst = prefix + '/share/man/man6/fceux.6'
-man_net_dst = prefix + '/share/man/man6/fceux-net-server.6'
+

 share_src = 'output/'
 share_dst = prefix + '/share/fceux/'
@@ -230,12 +230,12 @@
 desktop_dst = prefix + '/share/applications/'

 env.Install(prefix + "/bin/", fceux)
-env.Install(prefix + "/bin/", "fceux-net-server")
+
 # TODO:  Where to put auxlib on "scons install?"
 env.Alias('install', env.Command(auxlib_inst_dst, auxlib_src, [Copy(auxlib_inst_dst, auxlib_src)]))
 env.Alias('install', env.Command(share_dst, share_src, [Copy(share_dst, share_src)]))
 env.Alias('install', env.Command(man_dst, man_src, [Copy(man_dst, man_src)]))
-env.Alias('install', env.Command(man_net_dst, man_net_src, [Copy(man_net_dst, man_net_src)]))
+
 env.Alias('install', env.Command(image_dst, image_src, [Copy(image_dst, image_src)]))
 env.Alias('install', env.Command(desktop_dst, desktop_src, [Copy(desktop_dst, desktop_src)]))
 env.Alias('install', (prefix + "/bin/"))
Common subdirectories: fceux-2.2.2-orig/src and fceux-2.2.2/src