diff --git a/Makefile.am b/Makefile.am
index cc15338..1752f18 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -45,7 +45,7 @@ pkgconfig_DATA = libsecp256k1.pc
 libsecp256k1_la_SOURCES = src/secp256k1.c
 libsecp256k1_la_CPPFLAGS = -I$(top_srcdir)/include $(SECP_INCLUDES)
 libsecp256k1_la_LIBADD = $(SECP_LIBS)
-
+libsecp256k1_la_LDFLAGS = -no-undefined

 noinst_PROGRAMS =
 if USE_BENCHMARK
diff --git a/configure.ac b/configure.ac
index 3dc1829..45cd274 100644
--- a/configure.ac
+++ b/configure.ac
@@ -7,7 +7,7 @@ AH_TOP([#ifndef LIBSECP256K1_CONFIG_H])
 AH_TOP([#define LIBSECP256K1_CONFIG_H])
 AH_BOTTOM([#endif /*LIBSECP256K1_CONFIG_H*/])
 AM_INIT_AUTOMAKE([foreign subdir-objects])
-LT_INIT
+LT_INIT([win32-dll])

 dnl make the compilation flags quiet unless V=1 is used
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
