diff -u -r avrdude-6.2-orig/configure.ac avrdude-6.2/configure.ac
--- avrdude-6.2-orig/configure.ac	2015-11-28 09:45:32.828765400 -0800
+++ avrdude-6.2/configure.ac	2015-11-29 12:31:19.856337900 -0800
@@ -34,7 +34,7 @@
 AC_CONFIG_HEADERS(ac_cfg.h)
 AC_CONFIG_MACRO_DIR([m4])
 
-LT_INIT()
+LT_INIT([win32-dll])
 
 # Checks for programs.
 AC_PROG_CC
@@ -185,6 +185,8 @@
 AC_CHECK_HEADERS([fcntl.h sys/ioctl.h sys/time.h termios.h unistd.h])
 AC_CHECK_HEADERS([ddk/hidsdi.h],,,[#include <windows.h>
 #include <setupapi.h>])
+AC_CHECK_HEADERS([hidsdi.h],,,[#include <windows.h>
+#include <setupapi.h>])
 
 
 # Checks for typedefs, structures, and compiler characteristics.
@@ -202,9 +204,12 @@
 case $target in
         *-*-mingw32* | *-*-cygwin* | *-*-windows*)
 		LIBHID="-lhid -lsetupapi"
-		if test $ac_cv_header_ddk_hidsdi_h = yes
+		if test x$ac_cv_header_ddk_hidsdi_h = xyes
 		then
 			HIDINCLUDE="#include <ddk/hidsdi.h>"
+		elif test x$ac_cv_header_hidsdi_h = xyes
+		then
+			HIDINCLUDE="#include <hidsdi.h>"
 		else
 			HIDINCLUDE="#include \"my_ddk_hidsdi.h\""
 		fi
diff -u -r avrdude-6.2-orig/Makefile.am avrdude-6.2/Makefile.am
--- avrdude-6.2-orig/Makefile.am	2015-11-28 09:45:32.758379800 -0800
+++ avrdude-6.2/Makefile.am	2015-11-28 10:28:26.752297500 -0800
@@ -138,7 +138,6 @@
 	linuxgpio.h \
 	linux_ppdev.h \
 	lists.c \
-	my_ddk_hidsdi.h \
 	par.c \
 	par.h \
 	pgm.c \
@@ -176,7 +175,7 @@
 	wiring.h \
 	wiring.c
 libavrdude_la_SOURCES = $(libavrdude_a_SOURCES)
-libavrdude_la_LDFLAGS = -version-info 1:0
+libavrdude_la_LDFLAGS = -no-undefined -version-info 1:0
 
 include_HEADERS = libavrdude.h
 
diff -u -r avrdude-6.2-orig/pickit2.c avrdude-6.2/pickit2.c
--- avrdude-6.2-orig/pickit2.c	2015-11-28 09:45:32.674308700 -0800
+++ avrdude-6.2/pickit2.c	2015-11-28 10:04:30.913297900 -0800
@@ -60,6 +60,8 @@
 #include <windows.h>
 #if defined(HAVE_DDK_HIDSDI_H)
 #  include <ddk/hidsdi.h>
+#elif defined(HAVE_HIDSDI_H)
+#  include <hidsdi.h>
 #else
 #  include "my_ddk_hidsdi.h"
 #endif
diff -u -r avrdude-6.2-orig/ser_avrdoper.c avrdude-6.2/ser_avrdoper.c
--- avrdude-6.2-orig/ser_avrdoper.c	2015-11-28 09:45:32.851250500 -0800
+++ avrdude-6.2/ser_avrdoper.c	2015-11-28 10:09:24.206699100 -0800
@@ -71,10 +71,12 @@
 
 #if defined(HAVE_DDK_HIDSDI_H)
 #  include <ddk/hidsdi.h>
+#elif defined (HAVE_HIDSDI_H)
+#  include <hidsdi.h>
 #else
 #  include "my_ddk_hidsdi.h"
 #endif
-#include <ddk/hidpi.h>
+#include <hidpi.h>
 
 #ifdef USB_DEBUG
 #define DEBUG_PRINT(arg)    printf arg
diff -u -r avrdude-6.2-orig/serbb_win32.c avrdude-6.2/serbb_win32.c
--- avrdude-6.2-orig/serbb_win32.c	2015-11-28 09:45:32.912837800 -0800
+++ avrdude-6.2/serbb_win32.c	2015-11-28 10:38:03.674831400 -0800
@@ -309,7 +309,7 @@
                 return -1;
 	}
         avrdude_message(MSG_DEBUG, "%s: ser_open(): opened comm port \"%s\", handle 0x%x\n",
-                        progname, port, (int)hComPort);
+                        progname, port, hComPort);
 
         pgm->fd.pfd = (void *)hComPort;
 
@@ -327,7 +327,7 @@
 		CloseHandle (hComPort);
 	}
         avrdude_message(MSG_DEBUG, "%s: ser_close(): closed comm port handle 0x%x\n",
-                                progname, (int)hComPort);
+                                progname, hComPort);
 
 	hComPort = INVALID_HANDLE_VALUE;
 }
diff -u -r avrdude-6.2-orig/windows/Makefile.am avrdude-6.2/windows/Makefile.am
--- avrdude-6.2-orig/windows/Makefile.am	2015-11-28 09:45:32.698747700 -0800
+++ avrdude-6.2/windows/Makefile.am	2015-11-28 10:01:45.066756000 -0800
@@ -24,11 +24,7 @@
 # This Makefile will only be used on windows based systems.
 #
 
-local_install_list = \
-	giveio.sys \
-	install_giveio.bat \
-	remove_giveio.bat \
-	status_giveio.bat
+local_install_list =
 
 EXTRA_DIST   = \
 	giveio.c \
